wallet/types/src/utxo_types.rs

```diff
@@ -29,6 +31,16 @@ pub enum UtxoType {
     DelegateStaking = 1 << 6,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u16)]
+pub enum UtxoState {
```

Comment:
Are these supposed to be mutually exclusive? If they are, then having them encoded as bit flags is a bad idea. Besides that, I think this structure is too forgiving and lacks checks. I don't have anything specific in mind, but can't we code some sanity verification within it, so that if something goes wrong we can detect it?

Reply:
not mutually exclusive, it is the same bit flag approach as the above one for UtxoType